OPHD/MicroPather/micropather.cpp

```diff
@@ -939,8 +939,8 @@ int MicroPather::Solve(void* startNode, void* endNode, std::vector< void* >* pat
 				PathNode* child = nodeCostVec[i].node;
 				float newCost = node->costFromStart + nodeCostVec[i].cost;
 
-				PathNode* inOpen = child->inOpen ? child : nullptr;
-				PathNode* inClosed = child->inClosed ? child : nullptr;
+				PathNode* inOpen = (child && child->inOpen) ? child : nullptr;
+				PathNode* inClosed = (child && child->inClosed) ? child : nullptr;

```

Comment:
I wonder if we should `throw`, rather than add a guard here. Technically the guard changes behaviour. Code that likely would have hard crashed before, may now continue to run with silently incorrect behaviour.

Reply:
If you look further down, the code already asserts when `child` is `nullptr`:

```cpp
MPASSERT(!(inOpen && inClosed));
```

Prior to this change, if child is `nullptr`, so will `inOpen and `inClosed` and as such, the MPASSERT check will pass and throw an assertion.

Reply:
That's hardly a valid check though. The `inOpen` and `inClosed` checked are derived from `child`, and they do so by dereferencing it. It `child` was `nullptr`, the program would have already crashed.

Maybe we should consider adding:
```cpp
MPASSERT(child);
```

It can be placed before the block that dereferences `child` to get the `inOpen` and `inClosed` values.

Once the assert is passed, we would no longer need to check individual expressions.

----

The check for both `inOpen` and `inClosed` is separate, and still potentially needed. It's possible a node could somehow end up being classified as being in both the open and closed sets simultaneously. Some algorithms are legitimated designed to work that way. This doesn't appear to be one of those algorithms.